lib/domain/model/extensions/xfile_extension.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:file_picker/file_picker.dart';
+import 'package:file_selector/file_selector.dart';
+import 'package:matrix/matrix.dart';
+
+extension XFileExtension on XFile {
+  Future<MatrixFile> toMatrixFile() async {
+    return MatrixFile.fromMimeType(
+      bytes: await readAsBytes(),
+      mimeType: mimeType,
+      name: name,
+      filePath: path,
+      sizeInBytes: await length(),
+    );
+  }
+
+  Future<PlatformFile> toPlatformFile() async {
+    return PlatformFile.fromMap({
+      'name': name,
+      'path': path,
+      'bytes': await readAsBytes(),
+      'size': await length(),
+    });
+  }
```

Comment:
we can't use files in linux ? what about stream, i think we use bytes only when we are desperate

Reply:
Yes we can use files but XFile is the type returned by file_selector and I wanted to reuse as much as possible the existing logic, so that's why I created those methods

Reply:
yeah its okay, but i shoudn't use `readAsBytes` method, you can use its path to read the data of file

Reply:
so you want to convert from FilePicker result right ?, FilePicker result also have Stream or File, so no need to convert to bytes